### PR TITLE
fix(Layer features): 20006, 19171 update layerFeaturesCollectionAtom to reatom3. Fix duplicated /search requests

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/FE-Update-layerFeaturesCollectionAtom-to-reatom3-20006
https://kontur.fibery.io/Tasks/Task/FE-DN-fetches-layers-feature-items-search-three-times-during-initialization-19171

## Release Notes

- **New Features**
  - Introduced a new function for fetching layer features, enhancing data retrieval capabilities.
  - Added improved error messages for loading layer features and when no features are found.
  - Added a new feature flag for the Layer Features Panel, allowing for configurable behavior.
  - Enhanced the Layer Features Panel configuration to specify if an enabled layer is required.

- **Bug Fixes**
  - Updated handling of feature states to prevent runtime errors when no feature is selected.

- **Refactor**
  - Streamlined state management and atom definitions for better maintainability.
  - Renamed functions for clarity and updated import paths to reflect directory restructuring.

- **Documentation**
  - Enhanced documentation for the Layer Features Panel, detailing configuration, data fetching, and rendering processes.
  - Improved localization for user feedback and error handling.
  - Added a new constant for minimum panel height.

### CODE REVIEW

File: src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts [hunk 1]
```diff
@@ -28,101 +26,52 @@ export const featuresPanelLayerId: string =
...
-            const panelData = transformFeaturesToPanelData(response);
-            dispatch(create('_setState', panelData));
-          });
-      }
-    });
-
-    return state;
-  },
-  'layerFeaturesCollectionAtom',
+export const currentFeatureIdAtom = atom<number | undefined>(
```
Comment (Natallia-Harshunova) [inline src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts]: I personally prefer to use `null` as initial state if that state is set explicitly

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts (hunk 1 above)
Comment (albaranau) [reply src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts]: I agree here. Changed to null

File: src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx [hunk 2]
```diff
@@ -31,9 +31,9 @@ import type { FeatureCardCfg } from '../CardElements';
 import type { Bbox } from '~core/shared_state/currentMapPosition';
 
 export function LayerFeaturesPanel() {
-  const [currentFeatureId, { set: setCurrentFeatureId }] = useAtom(currentFeatureIdAtom);
+  const [currentFeatureId] = useAtom(currentFeatureIdAtom);
   const onCurrentChange = (id: number, feature: FeatureCardCfg) => {
-    setCurrentFeatureId(id);
+    currentFeatureIdAtom(store.v3ctx, id);
```
Comment (vkozio) [inline src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx]: no need to do it that way inside React component. use setter from atom hook https://www.reatom.dev/package/npm-react/#useatom

**Review by vkozio — CHANGES_REQUESTED** [review]

File: src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx (hunk 2 above)
Comment (albaranau) [reply src/features/layer_features_panel/components/LayerFeaturesPanel/index.tsx]: Thanks, forgot about the setter function! Fixed

**Review by vkozio — APPROVED** [review]

**Pull request merged**